adapter/internal/oasparser/envoyconf/routes_with_clusters.go

```diff
@@ -847,6 +959,66 @@ func createRoute(params *routeCreateParams) *routev3.Route {
 				IdleTimeout:       ptypes.DurationProto(time.Duration(config.Envoy.Upstream.Timeouts.RouteIdleTimeoutInSeconds) * time.Second),
 			},
 		}
+		if config.Envoy.RateLimit.Enabled &&
+			((ratelimitLevel == RateLimitPolicyOperationLevel && isRLPolicyAvailable) || ratelimitLevel == RateLimitPolicyAPILevel) {
```

Comment:
If we do https://github.com/wso2/product-microgateway/pull/3142/files#r1053185403 we can simplify this condition as below

```suggestion
		if config.Envoy.RateLimit.Enabled && isRLPolicyAvailable) {
```